tensorflow_graphics/rendering/camera/perspective.py

```diff
@@ -201,40 +201,45 @@ def intrinsics_from_matrix(matrix, name=None):
       "perspective_intrinsics_from_matrix".
 
   Returns:
-    Tuple of two tensors, each one of shape `[A1, ..., An, 2]`. The first
-    tensor represents the focal length, and the second one the principle point.
+    Tuple of three tensors, the first two of shape `[A1, ..., An, 2]` and
+    the third of shape `[A1, ..., An, 1]`. The first tensor represents the
+    focal length, and the second one the principle point and the third one
+    represents the skew coefficient.
 
   Raises:
     ValueError: If the shape of `matrix` is not supported.
   """
-  with tf.compat.v1.name_scope(name, "perspective_intrinsics_from_matrix",
-                               [matrix]):
+  with tf.compat.v1.name_scope(name, "perspective_intrinsics_from_matrix", [matrix]):
     matrix = tf.convert_to_tensor(value=matrix)
 
     shape.check_static(
         tensor=matrix,
         tensor_name="matrix",
         has_rank_greater_than=1,
-        has_dim_equals=((-1, 3), (-2, 3)))
+        has_dim_equals=((-1, 3), (-2, 3)),
+    )
 
     fx = matrix[..., 0, 0]
     fy = matrix[..., 1, 1]
     cx = matrix[..., 0, 2]
     cy = matrix[..., 1, 2]
+    skew = matrix[..., 0, 1]
     focal = tf.stack((fx, fy), axis=-1)
     principal_point = tf.stack((cx, cy), axis=-1)
-  return focal, principal_point
+    skew = tf.expand_dims(skew, axis=0)
 
+    return focal, principal_point, skew
 
-def matrix_from_intrinsics(focal, principal_point, name=None):
+
+def matrix_from_intrinsics(focal, principal_point, skew=(0.0,), name=None):
```

Comment:
Given that the shape of skew has to be coerced to match focal and prinicipal_point anyway, it probably makes sense for the default value to be just the scalar "0.0", not the tuple "(0.0,)".